### PR TITLE
Add support for GeoJSON on Google Maps

### LINKED ISSUE #60: Add GeoJSON support (Google Maps)

### PR DESCRIPTION
**Pull request by szdziedzic**
- [x] Add GeoJSON support on Google Maps (Android)
- [x] Add GeoJSON support on Google Maps (iOS)
- [x] GeoJSON styling on Google Maps (Android)
- [x] Add support for default style on Android
- [x] Add support for default style on iOS
- [ ] GeoJSON styling on Google Maps (iOS)
- [x] Add example for GeoJSON feature
- [x] Support for "named" colors for Polygons, Polylines and GeoJSONs in our library

### CODE REVIEW

**Comment by szdziedzic** [comment]
I you only want you can take a look at this PR and give me your first reviews 😄.

Unfortunately I'm still working on styling on Google Maps for iOS. Generally speaking styling works much better on Android and it's very tricky on iOS, so it can take me few moments to add our own utils compensating for lack of functionalities in iOS version of library.

Let me know what do you think 😄

File: expo-maps/android/src/main/java/expo/modules/maps/googleMaps/GoogleMapsGeoJsons.kt
Comment (lukmccall) [inline expo-maps/android/src/main/java/expo/modules/maps/googleMaps/GoogleMapsGeoJsons.kt]:
```suggestion
      polygonStyle.strokePattern = layer.defaultPolygonStyle.strokePattern
```

File: expo-maps/android/src/main/java/expo/modules/maps/PropsData.kt
Comment (lukmccall) [inline expo-maps/android/src/main/java/expo/modules/maps/PropsData.kt:125]: I have a question here. Why did you use `Float` as a type here? I'm curious, cause the more natural representation of the js number in `Kotlin` will be `Double`.

File: expo-maps/android/src/main/java/expo/modules/maps/interfaces/GeoJsons.kt
Comment (lukmccall) [inline expo-maps/android/src/main/java/expo/modules/maps/interfaces/GeoJsons.kt]:
```suggestion
  fun setGeoJsons(geoJsonObjects: Array<GeoJsonObject>)
```

File: expo-maps/android/src/main/java/expo/modules/maps/PropsData.kt
Comment (szdziedzic) [reply expo-maps/android/src/main/java/expo/modules/maps/PropsData.kt:125]: I used `Float` because, library expects `strokeWidth` argument to be `float` type. I agree that `Double` is more natural representation of the JS number in Kotlin. Do you recommend using `Double` type for the prop and then parsing it to the `float` type when assigning our prop to library's `strokeWidth`?

Comment (lukmccall) [reply expo-maps/android/src/main/java/expo/modules/maps/PropsData.kt:125]: In that case no, if the library below uses `float`, let's stick to it ;)

File: expo-maps/build/GeoJson.d.ts
Comment (p-karcz) [inline expo-maps/build/GeoJson.d.ts:139]: I think there's no need to include an example geojson input file here but if you think it would be beneficial then leave it and feel free to resolve.

File: expo-maps/src/GeoJson.ts
Comment (p-karcz) [inline expo-maps/src/GeoJson.ts:145]: same as above

**Review by p-karcz — APPROVED** [review]
LGTM, good job!

File: expo-maps/build/GeoJson.d.ts
Comment (szdziedzic) [reply expo-maps/build/GeoJson.d.ts:139]: I'm open to discuss it, but I assumed that it might be useful for the user to see the example and therefore be able to easier understand our functionalities 🤔 What do you think? Do you find it helpful?

**Comment by szdziedzic** [comment]
I think I might add styling for individual GeoJSON features in separate PR to not hold this one any longer.

File: expo-maps/build/GeoJson.d.ts
Comment (p-karcz) [reply expo-maps/build/GeoJson.d.ts:139]: I thought there is a standard for GeoJson. You're right, the example will be helpful

File: expo-maps/src/Map.tsx
Comment (p-karcz) [inline expo-maps/src/Map.tsx]: I think it could be extracted to a private method so it would be more readable, but it's just styling so I leave it up to you

**Review by p-karcz — APPROVED** [review]

File: expo-maps/src/Map.tsx
Comment (szdziedzic) [reply expo-maps/src/Map.tsx]: Sounds like good idea 👍

**Review by ellwiii — APPROVED** [review]

File: expo-maps/android/src/main/java/expo/modules/maps/googleMaps/GoogleMapsGeoJsons.kt
Comment (lukmccall) [inline expo-maps/android/src/main/java/expo/modules/maps/googleMaps/GoogleMapsGeoJsons.kt]:
```suggestion
    polygonStyle.strokeColor = if (feature.hasProperty("strokeColor")) {
      colorStringToARGBInt(feature.getProperty("strokeColor"))
    } else {
      layer.defaultPolygonStyle.strokeColor
    }
```
What do you think? In my opinion that looks better, but it is up to you.

File: expo-maps/android/src/main/java/expo/modules/maps/googleMaps/GoogleMapsUtils.kt
Comment (lukmccall) [inline expo-maps/android/src/main/java/expo/modules/maps/googleMaps/GoogleMapsUtils.kt:82]:
```suggestion
  return jointTypeString.toIntOrNull() ?: when (jointTypeString) {
    "miter" -> JointType.DEFAULT
    "bevel" -> JointType.BEVEL
    "round" -> JointType.ROUND
    else -> throw IllegalArgumentException("Illegal joint type name: $jointTypeString")
  }
```

**Review by lukmccall — APPROVED** [review]

File: expo-maps/android/src/main/java/expo/modules/maps/googleMaps/GoogleMapsGeoJsons.kt
Comment (szdziedzic) [reply expo-maps/android/src/main/java/expo/modules/maps/googleMaps/GoogleMapsGeoJsons.kt]: It also looks better in my opinion, thank you

**Pull request merged**
